cypress/support/pages/admin/pageAdministration.js

```diff
@@ -643,42 +663,73 @@ class Management {
 
 	clickOnAddUserInFAB(role) {
 		let addUserButtonInFAB =
-			role === "student" ? Management.#addStudentButton : Management.#addTeacherButton;
+			role === "student"
+				? Management.#addStudentButton
+				: Management.#addTeacherButton;
 		cy.get(addUserButtonInFAB).click({ force: true });
 	}
 
 	fillUserCreationForm(forename, surname, baseEmail) {
 		const randomNumber = new Date().getTime() + Math.floor(Math.random() * 1000);
 		const uniqueEmail = randomNumber + baseEmail;
 
-		// Store the generated unique email as an alias for later use
+		// store the generated unique email as an alias for later use
 		cy.wrap(uniqueEmail).as("uniqueEmail");
-
-		// Log the generated unique email
 		cy.log("Generated Unique Email:", uniqueEmail);
 
-		// Fill in the form with the generated email and other details
+		// fill in the form with the generated email and other details
 		cy.get(Management.#firstNameCreationForm).type(forename);
 		cy.get(Management.#lastNameCreationForm).type(surname);
 		cy.get(Management.#emailCreationForm).type(uniqueEmail);
 
-		// Setting the birth date to 17 years ago in the form
+		// setting the birth date to 17 years ago in the form for student user
 		cy.get("body").then((body) => {
 			if (body.find(Management.#birthDateFieldCreateStudent).length) {
-				// For Student
 				const birthDate = new Date();
 				birthDate.setFullYear(birthDate.getFullYear() - 17);
-				cy.get(Management.#birthDateFieldCreateStudent).type(
-					birthDate.toISOString().split("T")[0],
-					100
-				);
+
+				// YYYY-MM-DD for typing into the input
+				// note: toISOString() returns date in UTC, so we split and take only the date part
+				// this ensures that the date is correctly interpreted regardless of local timezone
+				const isoDate = birthDate.toISOString().split("T")[0];
+				const formattedBirthDate = `${String(birthDate.getDate()).padStart(2, "0")}.${String(birthDate.getMonth() + 1).padStart(2, "0")}.${birthDate.getFullYear()}`; // DD.MM.YYYY for table check
```

Comment:
this long string can be avoided with `DateTimeFormat` set it for 2 digit for day, month and year will be probably numeric.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat

Reply:
method is now refactored.